### PR TITLE
Setup tests

### PR DESCRIPTION
**Pull request by mqharris**

### CODE REVIEW

File: .github/workflows/R-CMD-check.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# For help debugging build failures open an issue on the RStudio community with the 'github-actions' tag.
```
Comment (mqharris) [inline .github/workflows/R-CMD-check.yaml:1]: this was all auto generated 
except for `on: [push, pull_request]` We can dial this in where ever we like.

File: R/utils.R [hunk 2]
```diff
@@ -18,20 +20,25 @@ get_vrd_key <- function() {
...
+#'
 #' @return vrd_api class
 #' @export
 #'
 #' @examples
 #' \dontrun{
 #' call_vrd_api()
 #' }
-call_vrd_api <- function(url_, query, limit = NULL){
+call_vrd_api <- function(url_, query, limit = NULL, api_key=get_vrd_key()){
```
Comment (mqharris) [inline R/utils.R:35]: defaults to calling the get api key function

sane default behaviour, and makes it easy to test

File: tests/testthat/test-utils.R [hunk 3]
```diff
@@ -0,0 +1,13 @@
+test_that("bad api key correctly fails", {
+  make <- "Mazda"
+  url_ <- "https://vrdb-tc-apicast-production.api.canada.ca/eng/vehicle-recall-database/v1/recall/make-name/"
+  url_ <- paste(url_, toString(make), "/count", sep = "")
+  expect_error(call_vrd_api(url_,make, api_key="test"))
+})
+
+test_that("call_rd_api() returns the correct structure",{
```
Comment (mqharris) [inline tests/testthat/test-utils.R:8]: tests the api call with out having to set an api key

there are a few default/test api urls outlined in the word doc describing api on the website

File: .github/workflows/R-CMD-check.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    runs-on: macOS-latest
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: r-lib/actions/setup-r@v1
+      - name: Install dependencies
+        run: |
+          install.packages(c("remotes", "rcmdcheck"))
+          remotes::install_deps(dependencies = TRUE)
```
Comment (WraySmith) [inline .github/workflows/R-CMD-check.yaml:18]: so this automatically reads from the description file, or do we need to put the dependencies in here?

File: R/utils.R [hunk 5]
```diff
@@ -4,7 +4,9 @@
 #' @export
 #'
 #' @examples
+#' \dontrun{
 #' get_vrd_key()
```
Comment (WraySmith) [inline R/utils.R:8]: why aren't we running this?

Comment (WraySmith) [reply R/utils.R:8]: dealing with later

**Review by WraySmith — APPROVED** [review]

**Pull request merged**